eng/SourceBuild.props

```diff
@@ -0,0 +1,119 @@
+          BeforeTargets="RunInnerSourceBuildCommand">
+
+    <ItemGroup>
+      <SourceBuildPatchFile Include="$(RepositoryEngineeringDir)source-build-patches\*.patch" />
+    </ItemGroup>
+
+    <Exec
+      Command="git apply --ignore-whitespace --whitespace=nowarn &quot;%(SourceBuildPatchFile.FullPath)&quot;"
+      WorkingDirectory="$(InnerSourceBuildRepoRoot)"
+      Condition="'@(SourceBuildPatchFile)' != ''" />
+  </Target>
+
+  <Target Name="CategorizeRuntimeSupplementalArtifacts"
+          BeforeTargets="GetCategorizedIntermediateNupkgContents">
+    <ItemGroup>
+      <!--
+        Order matters in some situations: put more specific patterns first, so
+        they are the ones "picked" during item-name-only deduplication.
+      -->
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-hostfxr-internal-*.tar.gz" Category="hostfxr-internal" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-runtime-internal-*.tar.gz" Category="runtime-internal" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-runtime-*.tar.gz" Category="runtime" />
+
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-crossgen2-*.tar.gz" Category="Crossgen2Pack.Archive" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-apphost-pack-*.tar.gz" Category="HostPack.Archive" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-targeting-pack-*.tar.gz" Category="RefPack.Archive" />
+
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\Microsoft.NETCore.App.Crossgen2.*.nupkg" Category="Crossgen2Pack" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\Microsoft.NETCore.App.Host.*.nupkg" Category="HostPack" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\Microsoft.NETCore.App.Ref.*.nupkg" Category="RefPack" />
+      <IntermediateNupkgArtifactFile Include="$(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\Microsoft.NETCore.App.Runtime.*.nupkg" Category="RuntimePack" />
+
+      <IntermediateNupkgArtifactFile
+        Include="
+          $(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\dotnet-ilverify.*.nupkg;
+          $(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\Microsoft.NET.Sdk.IL.*.nupkg;
+          $(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\*Microsoft.NETCore.ILAsm.*.nupkg;
+          $(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\*Microsoft.NETCore.ILDAsm.*.nupkg;
+          $(CurrentRepoSourceBuildArtifactsPackagesDir)Shipping\*Microsoft.NETCore.TestHost.*.nupkg;
+          "
+        Category="CoreCLR" />
```

Comment:
I would like to better understand why these categories are necessary? We don't have a categorization today in the repo and hard coding all the produced assets for source build isn't really maintainable.

Reply:
Here is the background on this - https://github.com/dotnet/source-build/issues/2034 and https://github.com/dotnet/source-build/blob/main/Documentation/planning/arcade-powered-source-build/intermediate-nupkg.md#too-large.  The driving reason a single intermediate NuGet package is too large to publish.

Reply:
Per our discussion earlier today I updated the split to be chunkier until the Arcade changes are in-place to handle splitting the artifacts automatically.  There are only three packages produced now, two being the supplemental ones.

371M Microsoft.SourceBuild.Intermediate.runtime.runtime.linux-x64.6.0.0-dev.nupkg
319M Microsoft.SourceBuild.Intermediate.runtime.linux-x64.6.0.0-dev.nupkg
357M Microsoft.SourceBuild.Intermediate.runtime.Crossgen2Pack.linux-x64.6.0.0-dev.nupkg